src/kOS.Safe/Serialization/Formatter.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace kOS.Safe.Serialization
+{
+    public interface Formatter
+    {
+        string Write(IDictionary<object, object> value);
+        IDictionary<object, object> Read(string input);
```

Comment:
we only have 2 implementations of this interface and we already have segmentation in that `TerminalFormatter` cant read. we should split the interface with a reader and a writer